### PR TITLE
Add stair corners rework

### PR DESCRIPTION
**Pull request by TumeniNodes**
This is to replace https://github.com/minetest/minetest_game/pull/1814 (if someone could close #1814 please, as it is a tumor) I apologize for all the troubles with #1814.

This finishes the work MarkuBu has done, with the addition of the models provided by GreenXenith.
It also adds two stair set materials: Clay and Tin

### CODE REVIEW

File: mods/stairs/init.lua [hunk 1]
```diff
@@ -15,7 +15,7 @@ minetest.register_alias("stairs:slab_pinewood", "stairs:slab_pine_wood")
 
 -- Get setting for replace ABM
 
-local replace = minetest.settings:get_bool("enable_stairs_replace_abm")
+local replace = minetest.setting_getbool("enable_stairs_replace_abm")
```
Comment (SmallJoker) [inline mods/stairs/init.lua]: Rebase error. Should be `minetest.settings:get_bool`

File: mods/stairs/init.lua [hunk 2]
```diff
@@ -46,24 +46,12 @@ end
...
-		if type(image) == "string" then
-			stair_images[i] = {
-				name = image,
-				backface_culling = true,
-			}
-		elseif image.backface_culling == nil then -- override using any other value
-			stair_images[i] = table.copy(image)
-			stair_images[i].backface_culling = true
-		end
-	end
```
Comment (SmallJoker) [inline mods/stairs/init.lua]: Rebase error too. Please undo.

File: mods/stairs/init.lua [hunk 3]
```diff
@@ -228,15 +216,6 @@ function stairs.register_slab(subname, recipeitem, groups, images, description,
...
 
-		-- Use 2 slabs to craft a full block again (1:1)
-		minetest.register_craft({
-			output = recipeitem,
-			recipe = {
-				{'stairs:slab_' .. subname},
-				{'stairs:slab_' .. subname},
-			},
-		})
-
```
Comment (SmallJoker) [inline mods/stairs/init.lua]: Here too.

File: mods/stairs/init.lua [hunk 4]
```diff
@@ -276,13 +255,135 @@ if replace then
...
+		on_place = function(itemstack, placer, pointed_thing)
+			if pointed_thing.type ~= "node" then
+				return itemstack
+			end
+
+			return rotate_and_place(itemstack, placer, pointed_thing)
+		end,
+	})
+
+		if recipeitem then
```
Comment (SmallJoker) [inline mods/stairs/init.lua]: Please repair the whole indent block here.

File: mods/stairs/init.lua [hunk 5]
```diff
@@ -276,13 +255,135 @@ if replace then
...
+		end
+	end
+end
+
+function stairs.register_stair_outer(subname, recipeitem, groups, images, description, sounds)
+	groups.stair = 1
+	minetest.register_node(":stairs:stair_outer_" .. subname, {
+		description = description,
+		drawtype = "mesh",
+        mesh = "stairs_stair_outer.obj",
```
Comment (SmallJoker) [inline mods/stairs/init.lua]: Missing indent.

File: mods/stairs/init.lua [hunk 6]
```diff
@@ -550,6 +651,16 @@ stairs.register_stair_and_slab(
...
 
 stairs.register_stair_and_slab(
+	"clay",
+	"default:clay",
+	{cracky = 3},
+	{"default_clay.png"},
+	"Clay Stair",
+	"Clay Slab",
+	default.node_sound_stone_defaults()
+)
```
Comment (C1ffisme) [inline mods/stairs/init.lua]: I disagree with adding new stair types in the same pull request. Make Clay and Diamond Block Stairs and Slabs be a different pull request.

Comment (TumeniNodes) [inline mods/stairs/init.lua]: Don't worry about it. Not gonna happen anyway.
Maybe someone who is an experienced coder/programmer will do this someday (probably take them roughly ten minutes to do)
I'm not bothering, I'm done with this, enjoy

**Comment by TumeniNodes** [comment]
Wow, hey look, I learned how to close a PR... amazing.
